### PR TITLE
OCPBUGS-23511: Remove IAM path when applying control-plane-operator ARN to VPC Endpoint Service allowed principals

### PR DESCRIPTION
**Pull request by mjlshen**
**What this PR does / why we need it**:
Confusingly, AWS arns that include a path are not valid IAM principals, the path needs to be removed before they are applied to VPC Endpoint Service additional allowed principals.

**Which issue(s) this PR fixes**:
Fixes OCPBUGS-23511

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [x] This change includes unit tests.

/area hypershift-operator

### CODE REVIEW

File: hypershift-operator/controllers/platform/aws/controller.go [hunk 1]
```diff
@@ -642,3 +648,19 @@ func (r *AWSEndpointServiceReconciler) controlPlaneOperatorRoleARN(ctx context.C
 	}
 	return hc.Spec.Platform.AWS.RolesRef.ControlPlaneOperatorARN, nil
 }
+
+func (r *AWSEndpointServiceReconciler) controlPlaneOperatorRoleARNWithoutPath(ctx context.Context, hc *hyperv1.HostedCluster) (string, error) {
```
Comment (csrwng) [inline hypershift-operator/controllers/platform/aws/controller.go:652]: It doesn't look like `ctx` is used

Comment (mjlshen) [reply hypershift-operator/controllers/platform/aws/controller.go:652]: It isn't indeed - I was kind modeling it off of https://github.com/openshift/hypershift/pull/3215/files#diff-703bbd22baf7c689e315f5c561efed7afea4f14d0ade48d66af784206a0bc421R645 

```go
func (r *AWSEndpointServiceReconciler) controlPlaneOperatorRoleARN(ctx context.Context, hc *hyperv1.HostedCluster) (string, error)
```

Which similarly has an unused context, but am happy to remove this

Comment (mjlshen) [reply hypershift-operator/controllers/platform/aws/controller.go]: Removed both contexts :D

**Comment by andreadecorte** [comment]
lgtm

File: hypershift-operator/controllers/platform/aws/controller.go [hunk 2]
```diff
@@ -56,7 +57,11 @@ type AWSEndpointServiceReconciler struct {
 
 	// controlPlaneOperatorRoleARNFn determines the control plane operator role given a hosted cluster
 	// This is used to stub the function in unit tests
-	controlPlaneOperatorRoleARNFn func(context.Context, *hyperv1.HostedCluster) (string, error)
+	controlPlaneOperatorRoleARNFn func(*hyperv1.HostedCluster) (string, error)
```
Comment (enxebre) [inline hypershift-operator/controllers/platform/aws/controller.go:60]: can't this be removed now? it's not being consumed

File: hypershift-operator/controllers/platform/aws/controller.go [hunk 3]
```diff
@@ -56,7 +57,11 @@ type AWSEndpointServiceReconciler struct {
...
+
+	// controlPlaneOperatorRoleWithoutPathFn excludes the IAM path from an ARN, which is needed when adding the CPO
+	// IAM role to AWS IAM trust policies, namely the AWS VPC Endpoint Service allowed principals policy.
```
Comment (enxebre) [inline hypershift-operator/controllers/platform/aws/controller.go:63]: do we really need this stub?
can we not just have a `func controlPlaneOperatorRoleARNWithoutPath()` without receiver and invoke it inline?

Comment (mjlshen) [reply hypershift-operator/controllers/platform/aws/controller.go:63]: I thought the stub was strange too - I'll refactor it away into a standalone function

**Pull request merged**